SWLOR.Game.Server/Feature/ItemDefinition/HarvesterItemDefinition.cs

```diff
@@ -91,11 +93,17 @@ private void Harvester(string tag, int requiredLevel, params string[] validOreTa
                     if (resourceCount <= 0)
                     {
                         DestroyObject(target);
+
+                        // DestroyObject bypasses the OnDeath event, and removes the object so we can't send events.
```

Comment:
I overrode this DestroyObject call to do something for the AI already in Core/Extensions/DestroyObjectExtension.cs

If this is the case, I think I'd prefer to add something in that extension method to look at the OnDeath script and fire it, if it exists. That way we can do it in one place and not have to remember to do it any time we destroy a placeable.

Can you think of any issues with this approach? I know it's outside normal NWN mechanics but it's not intuitive to me the script doesn't fire.

Reply:
The problem with DestroyObject is that it... destroys the object.  So there's nothing left to run an event against (i.e. OBJECT_SELF won't work any more).  OnDeath happens logically *before* DestroyObject, and the object isn't actually destroyed until after all death processing is complete.

I agree my solution here is inelegant.  The best option would be to make EffectDeath() work on Placeables, I think - and then to apply that effect rather than calling DestroyObject.  Actually, checking the Lexicon, it looks like it already does - 'When it is applied to doors or placeable objects, they are destroyed and play a shatter animation.' - so I'll test using that instead.

Reply:
Good news - this worked.  Bad news - it requires toggling the plot flag off, so it's still replacing one line with two lines for placeable destruction.  Commit added.